File: computing/interview/design-patterns/creational/singleton/java/SingletonDemo.java
package interview.designpatterns.creational.singleton;

public enum Logger {
    INSTANCE;

    public void log(String message) {
        System.out.printf("[LOG] %s%n", message);
    }

    public static void main(String[] args) {
        Logger.INSTANCE.log("Starting process");
        Logger.INSTANCE.log("Process completed");
    }
}

File: computing/interview/design-patterns/creational/singleton/csharp/SingletonDemo.cs
using System;

namespace Interview.DesignPatterns.Creational.Singleton
{
    public sealed class Logger
    {
        private static readonly Lazy<Logger> _instance = new(() => new Logger());

        private Logger()
        {
        }

        public static Logger Instance => _instance.Value;

        public void Log(string message)
        {
            Console.WriteLine($"[LOG] {message}");
        }
    }

    public static class Demo
    {
        public static void Main()
        {
            Logger.Instance.Log("Starting process");
            Logger.Instance.Log("Process completed");
        }
    }
}
